internal/pkg/addon/package.go

```diff
@@ -273,8 +281,129 @@ type s3Object struct {
 }
 
 func (a *Addons) uploadAddonAsset(assetPath string, forceZip bool) (s3Object, error) {
+	// make path absolute from wsPath
+	if !path.IsAbs(assetPath) {
+		assetPath = path.Join(a.wsPath, assetPath)
+	}
+
+	info, err := a.fs.Stat(assetPath)
+	if err != nil {
+		return s3Object{}, err
+	}
+
+	var asset asset
+	if forceZip || info.IsDir() {
+		asset, err = a.zipAsset(assetPath)
+	} else {
+		asset, err = a.fileAsset(assetPath)
+	}
```

Comment:
nit: just a different way of writing it:
```go
var readAssetFn func(string) (asset, error) = a.fileAsset
if forceZip || info.IsDir() {
   readAssetFn = a.zipAsset
}
asset, err := readAssetFn(path)
```

Reply:
interesting! i like it!